ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/RequestConverter.java

```diff
@@ -21,34 +21,27 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class RequestConverter {
-    private final Ds3Request ds3Request;
-    private final ImmutableList<Arguments> requiredArguments;
-    private final ImmutableList<Arguments> optionalArguments;
-
+public final class RequestConverter {
     private static final Logger LOG = LoggerFactory.getLogger(RequestConverter.class);
 
-    private RequestConverter(final Ds3Request ds3Request) {
-        this.ds3Request = ds3Request;
-        this.requiredArguments = getRequiredArgs(ds3Request);
-        this.optionalArguments = getOptionalArgs(ds3Request);
-    }
+    private RequestConverter() {}
+
+    private final static RequestConverter requestConverter = new RequestConverter();
 
-    private Request convert() {
+    private Request convert(final Ds3Request ds3Request) {
```

Comment:
This can probably be made static removing the need to create an instance of RequestConverter